### PR TITLE
Code of conduct revision

### PR DESCRIPTION
**Pull request by jbmagination**
## PR changes
This change moves from the basic Contributor Covenant to an independent Definity Contributor Code of Conduct. This code of conduct revision has been adapted from both the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct) and the [Contributor Covenant](https://www.contributor-covenant.org/version/2/0/code_of_conduct.html). 

As of current, it's not much different. Here's the list of changes:
### Draft 1
- Name changed from `Contributor Covenant Code of Conduct` to `Definity Contributor Code of Conduct`
- The raw text of each paragraph is no longer separated every few words, and is now in full width
- Removed numerous newlines between header and following text
- A precursing text has been added, "This code of conduct outlines expectations for participation in Definity-managed open source communities, as well as steps for reporting unacceptable behavior. People violating this code of conduct may be banned from the community."
- The "Our Pledge" section has rewording of "an open, welcoming, inspiring, diverse, inclusive, healthy, and harassment-free community for everyone", adding of "caste, color", and a new sentence has been added, "We pledge to act, interact, and provide in ways that contribute to it."
- Examples of unacceptable behavior now also include "Disruptive behavior"
- The code of conduct scope now applies to actions taken outside of these spaces, *and* which have a negative impact on community health
- A change in violation reporting from Discord to email
- Both adapted codes of conduct have been attributed accordingly
### Draft 2
- Added license attribution for each code of conduct
- Added preamble header
- Removed remaining newlines between header and following text
### Draft 3
- Change from "Definity-managed open source communities" to "Definity-managed communities"
### Draft 4
- Shorten "Our Pledge"
- "Disruptive behavior" has been clarified
- Provide information as to what should be included in a report
- Make "Enforcement Guidelines" first sentence full width
### Draft 5
- Shorten "Our Pledge"
### Draft 6
- Shorten "Our Pledge"
- Update links
### Violation reporting
Right now, the email used to report violations is `support@definityteam.com`. This is probably not the best place to send violation reports, since the support inbox is more for solving issues with the things we make, and there's definitely going to be an extreme wait time.

Eventually, this should be its own email-- and as a bonus, we should probably get a dedicated group of people ~~who *cannot* interact with the community~~ alongside steps to verify with the reporter that the people involved with the issue are *not* on the team chosen to investigate. This way, it can be ensured that if people violate the code of conduct, it's not going to anyone who would be of said violation. That group can be for later, though, but it is *very* important.

### CODE REVIEW

**Comment by sudocode1** [comment]
Quite a bit of this feels like it's worded as if to be said towards youngwee schoolchildren. Otherwise all good.

**Comment by jbmagination** [comment]
If that's the case then that's something to take up with Contributor Covenant, because if that's the case then [this reads like it, too](https://www.contributor-covenant.org/version/2/0/code_of_conduct.html).

Will make a few final changes, then I'll consider it ready for review.

**Review by sudocode1 — APPROVED** [review]
LGTM. Ready to merge unless any other issues are brought up.

**Review by TimeCipher — CHANGES_REQUESTED** [review]
Line 6 starts as a well rounded sentence yet becomes too descriptive and develops into you simply listing diversifying features. Simply cutting this down will make it faster to read, keeping the readers attention. Otherwise it perfectly voices your goals in being equal.

File: CODE_OF_CONDUCT.md
Comment (sudocode1) [inline CODE_OF_CONDUCT.md:6]: After re-reading, I am in agreement with @TimeCipher, this just feels like a list of diversifying features & a list of "happy"-related buzz words. It should be changed to this, or with some relation to this. 
```suggestion
We as members, contributors, and leaders pledge to make participation in our community a welcoming and harassment-free place for everyone, regardless of age, body size, visible or invisible disability, or any other diversifying features. We pledge to act, interact, and provide in ways that contribute to it.
```

File: CODE_OF_CONDUCT.md
Comment (sudocode1) [inline CODE_OF_CONDUCT.md:22]: Simply adding "disruptive behaviour" is much too simple. It should be extended. 
```suggestion
* Disruptive behaviour; this includes harassment, unnecessary hostility and starting unneeded arguments.
```
While I understand this is meant to be concise, the way this is worded is much too simple.

File: CODE_OF_CONDUCT.md
Comment (sudocode1) [inline CODE_OF_CONDUCT.md:37]: This should provide information for what should be included within the report.
```suggestion
Instances of abusive, harassing, or otherwise unacceptable behavior may be reported to the community leaders responsible for enforcement at [support@definityteam.com](mailto:support@definityteam.com). All complaints will be reviewed and investigated promptly and fairly. Your report should include the username of the user, a screenshot or direct link to what you are complaining about and why you believe this is in violation of our Code of Conduct, and, if possible, what section of it.
```

**Review by sudocode1 — CHANGES_REQUESTED** [review]
I have re-read this commit after reading @TimeCipher's suggestion and I have a few changes that I think should be made before merge.

File: CODE_OF_CONDUCT.md
Comment (jbmagination) [reply CODE_OF_CONDUCT.md:6]: I can see the buzzwords part, but the features are specifically listed because if it's too vague, anything can fly there. You could argue a *lot* of things as being a "diversifying feature".

File: CODE_OF_CONDUCT.md
Comment (jbmagination) [reply CODE_OF_CONDUCT.md:22]: 👍

File: CODE_OF_CONDUCT.md
Comment (jbmagination) [reply CODE_OF_CONDUCT.md:37]: 👍

File: CODE_OF_CONDUCT.md
Comment (sudocode1) [reply CODE_OF_CONDUCT.md:6]: That is true, it really depends on if you want consistency with the README when deciding between using specifically my change or not. It should at least be shortened before merge.

**Comment by jbmagination** [comment]
Draft 4 should hopefully resolve your concerns, @sudocode1 @TimeCipher. Please feel free to request any more changes as to what should be modified, thank you both!

**Comment by jbmagination** [comment]
Actually, I'm not even quite satisfied with that. That can definitely be shortened more.

File: CODE_OF_CONDUCT.md
Comment (sudocode1) [reply CODE_OF_CONDUCT.md]: [Draft 5](https://github.com/DefinityTeam/.github/pull/2/commits/c295aa7a579d2d52a84f89b25ccb3321e27fde2f) resolves this. This is now concise and easier to read through.

**Review by sudocode1 — APPROVED** [review]
LGTM. Merge whenever, probably best to wait for more responses though.

**Pull request merged**
